framework/src/main/java/org/checkerframework/common/wholeprograminference/SceneToStubWriter.java

```diff
@@ -231,9 +231,9 @@ private static String formatArrayTypeImpl(ATypeElement scenelibRep, ArrayType ja
             return null;
         }
 
-        for (Map.Entry<InnerTypeLocation, ATypeElement> ite : e.innerTypes.entrySet()) {
-            InnerTypeLocation loc = ite.getKey();
-            if (loc.location.contains(TypePathEntry.ARRAY)) {
+        for (Map.Entry<List<TypePathEntry>, ATypeElement> ite : e.innerTypes.entrySet()) {
+            List<TypePathEntry> location = ite.getKey();
+            if (location.contains(new TypePathEntry(TypePath.ARRAY_ELEMENT, 0))) {
```

Comment:
Nit: creating a new object on every loop iteration feels inefficient.  How about making a static field for this?